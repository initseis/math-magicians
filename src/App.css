* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-size: 0.9rem;
}

.input-number {
  grid-area: input-number;
}

input {
  width: 100%;
  height: 100%;
  border: none;
  background-color: #858694;
  color: #fff;
  text-align: end;
  padding: 0 10px;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.ac {
  grid-area: ac;
}

.percentage {
  grid-area: percentage;
}

.division {
  grid-area: division;
}

.seven {
  grid-area: seven;
}

.eight {
  grid-area: eight;
}

.nine {
  grid-area: nine;
}

.by {
  grid-area: by;
}

.four {
  grid-area: four;
}

.five {
  grid-area: five;
}

.six {
  grid-area: six;
}

.less {
  grid-area: less;
}

.one {
  grid-area: one;
}

.two {
  grid-area: two;
}

.three {
  grid-area: three;
}

.plus {
  grid-area: plus;
}

.zero {
  grid-area: zero;
}

.dot {
  grid-area: dot;
}

.equal {
  grid-area: equal;
}

.grid-container {
  display: grid;
  grid-template-areas:
    "input-number input-number input-number input-number"
    "ac plus-less percentage division"
    "seven eight nine by"
    "four five six less"
    "one two three plus"
    "zero zero dot equal";
  background-color: #bababa;
  width: 300px;
  height: 250px;
  gap: 2px;
  margin: 200px auto auto auto;
}

button {
  width: 100%;
  height: 100%;
  border: none;
  cursor: pointer;
}

.grey-btn {
  background-color: #e0e0e0;
}

.orange-btn {
  background-color: #f5913e;
}
