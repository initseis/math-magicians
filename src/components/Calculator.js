/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import calculate from '../logic/calculate';
import operate from '../logic/operate';

const Calculator = () => {
  const [state, setState] = useState({
    total: '',
    next: null,
    operation: null,
  });

  const handleButton = (event) => {
    if (
      !(
        state.operation
        && (event.target.innerText === '+'
          || event.target.innerText === '-'
          || event.target.innerText === 'x'
          || event.target.innerText === 'รท'
          || event.target.innerText === '%')
      )
    ) {
      const obj = calculate(state, event.target.innerText);
      setState(obj);
      if (obj.total === null && obj.next === null && obj.operation === null) {
        document.querySelector('input').value = '';
      }
    }
  };

  return (
    <div className="grid-container">
      <div className="input-number">
        <input
          type="text"
          readOnly="readonly"
          value={state.next === null ? state.total : state.next}
        />
      </div>
      <div className="ac">
        <button type="button" className="grey-btn" onClick={handleButton}>
          AC
        </button>
      </div>
      <div className="plus-less">
        <button type="button" className="grey-btn" onClick={handleButton}>
          +/-
        </button>
      </div>
      <div className="percentage">
        <button type="button" className="grey-btn" onClick={handleButton}>
          %
        </button>
      </div>
      <div className="division">
        <button type="button" className="orange-btn" onClick={handleButton}>
          รท
        </button>
      </div>
      <div className="seven">
        <button type="button" className="grey-btn" onClick={handleButton}>
          7
        </button>
      </div>
      <div className="eight">
        <button type="button" className="grey-btn" onClick={handleButton}>
          8
        </button>
      </div>
      <div className="nine">
        <button type="button" className="grey-btn" onClick={handleButton}>
          9
        </button>
      </div>
      <div className="by">
        <button type="button" className="orange-btn" onClick={handleButton}>
          x
        </button>
      </div>
      <div className="four">
        <button type="button" className="grey-btn" onClick={handleButton}>
          4
        </button>
      </div>
      <div className="five">
        <button type="button" className="grey-btn" onClick={handleButton}>
          5
        </button>
      </div>
      <div className="six">
        <button type="button" className="grey-btn" onClick={handleButton}>
          6
        </button>
      </div>
      <div className="less">
        <button type="button" className="orange-btn" onClick={handleButton}>
          -
        </button>
      </div>
      <div className="one">
        <button type="button" className="grey-btn" onClick={handleButton}>
          1
        </button>
      </div>
      <div className="two">
        <button type="button" className="grey-btn" onClick={handleButton}>
          2
        </button>
      </div>
      <div className="three">
        <button type="button" className="grey-btn" onClick={handleButton}>
          3
        </button>
      </div>
      <div className="plus">
        <button type="button" className="orange-btn" onClick={handleButton}>
          +
        </button>
      </div>
      <div className="zero">
        <button type="button" className="grey-btn" onClick={handleButton}>
          0
        </button>
      </div>
      <div className="dot">
        <button type="button" className="grey-btn" onClick={handleButton}>
          .
        </button>
      </div>
      <div className="equal">
        <button type="button" className="orange-btn" onClick={handleButton}>
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
