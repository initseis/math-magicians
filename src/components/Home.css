.homeContainer {
  display: flex;
  flex-direction: column;
}

.homeContainer > p {
  font-size: 1.1rem;
  text-align: justify;
}

br {
  margin-bottom: 1rem;
}
