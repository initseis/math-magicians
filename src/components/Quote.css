h4 {
  text-align: justify;
  font-size: 1.5rem;
  color: #858694;
}

.quoteWrapper {
  padding: 8rem 25%;
}

h4:nth-child(3) {
  text-align: end;
}
