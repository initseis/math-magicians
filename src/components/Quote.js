import React from "react";

const Quote = () => {
  return <h2>Hello Quote</h2>;
};

export default Quote;
