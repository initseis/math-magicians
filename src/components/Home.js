import React from "react";
import "./Home.css";

const Home = () => {
  return (
    <div className="homeContainer">
      <h2>Welcome to our page!</h2>
      <br />
      <br />
      <p>
        Whether you are a student, business owner, engineer, scientist or just
        an average person working on your taxes, a calculator or scientific
        calculator is an essential tool to have on hand. A good calculator makes
        life easier and helps prevent errors. Maybe you are looking for help
        with finishing simple calculations like ringing up customers or figuring
        out the correct tip. On the other hand, maybe you are looking for a
        powerful graphing calculator capable of performing multiple functions.
        Whatever your needs, we're sure to have a calculator that will meet
        them.
      </p>
      <br />
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione, autem
        est odio cumque commodi voluptatem quam labore eaque officiis molestiae,
        dignissimos voluptate magni quasi exercitationem velit harum suscipit
        natus vel. Eum, soluta doloribus! Molestias dolorem corporis, soluta
        mollitia tempore vel totam quibusdam dolores aut, eaque itaque veritatis
        enim alias eum excepturi voluptates ea cupiditate doloribus. Aspernatur
        quasi voluptatem est sunt. Ducimus cupiditate dicta illo, fugit est unde
        eos maiores molestias! Quod iste nulla consectetur natus dolorum modi
        vero suscipit libero neque, expedita mollitia ea explicabo, placeat et,
        inventore iusto aspernatur?
      </p>
    </div>
  );
};

export default Home;
