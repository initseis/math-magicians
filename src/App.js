/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import {
  BrowserRouter as Router, Switch, Route, Link,
} from 'react-router-dom';
import './App.css';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quote from './components/Quote';

class App extends React.Component {
  render() {
    return (
      <Router>
        <div className="container">
          <nav>
            <Link className="logo" to="/">
              Math Magicians
            </Link>
            <ul>
              <li>
                <Link to="/math-magicians">Home</Link>
              </li>
              <li>
                <Link to="/math-magicians/calculator">Calculator</Link>
              </li>
              <li>
                <Link to="/math-magicians/quote">Quote</Link>
              </li>
            </ul>
          </nav>

          <Switch>
            <Route path="/math-magicians/calculator">
              <Calculator />
            </Route>
            <Route path="/math-magicians/quote">
              <Quote />
            </Route>
            <Route exact path="/math-magicians">
              <Home />
            </Route>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
