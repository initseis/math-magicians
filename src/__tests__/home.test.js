import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Calculator from '../components/Calculator';
import App from '../App';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Calculator', () => {
  test('renders App component', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: 2 }));
    // expect(screen.getByDisplayValue("1")).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toHaveValue('2');
    // screen.debug();

    // expect(screen.getByText(/Welcome/i)).toBeInTheDocument();
    // expect(screen.getByText("Welcome to our page!")).toBeInTheDocument();
  });
});

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// import Home from "../components/Home";

// describe("Home", () => {
//   test("renders App component", () => {
//     render(<Home />);

//     // screen.debug();

//     expect(screen.getByText(/Welcome/i)).toBeInTheDocument();
//     // expect(screen.getByText("Welcome to our page!")).toBeInTheDocument();
//   });
// });
