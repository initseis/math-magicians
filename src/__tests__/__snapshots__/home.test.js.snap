// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="homeContainer"
>
  <h2>
    Welcome to our page!
  </h2>
  <br />
  <br />
  <p>
    Whether you are a student, business owner, engineer, scientist or just an average person working on your taxes, a calculator or scientific calculator is an essential tool to have on hand. A good calculator makes life easier and helps prevent errors. Maybe you are looking for help with finishing simple calculations like ringing up customers or figuring out the correct tip. On the other hand, maybe you are looking for a powerful graphing calculator capable of performing multiple functions. Whatever your needs, we're sure to have a calculator that will meet them.
  </p>
  <br />
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione, autem est odio cumque commodi voluptatem quam labore eaque officiis molestiae, dignissimos voluptate magni quasi exercitationem velit harum suscipit natus vel. Eum, soluta doloribus! Molestias dolorem corporis, soluta mollitia tempore vel totam quibusdam dolores aut, eaque itaque veritatis enim alias eum excepturi voluptates ea cupiditate doloribus. Aspernatur quasi voluptatem est sunt. Ducimus cupiditate dicta illo, fugit est unde eos maiores molestias! Quod iste nulla consectetur natus dolorum modi vero suscipit libero neque, expedita mollitia ea explicabo, placeat et, inventore iusto aspernatur?
  </p>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="quoteWrapper"
>
  <h4>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  </h4>
  <br />
  <h4>
    - William Paul Thurston
  </h4>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="container"
>
  <nav>
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="homeContainer"
  >
    <h2>
      Welcome to our page!
    </h2>
    <br />
    <br />
    <p>
      Whether you are a student, business owner, engineer, scientist or just an average person working on your taxes, a calculator or scientific calculator is an essential tool to have on hand. A good calculator makes life easier and helps prevent errors. Maybe you are looking for help with finishing simple calculations like ringing up customers or figuring out the correct tip. On the other hand, maybe you are looking for a powerful graphing calculator capable of performing multiple functions. Whatever your needs, we're sure to have a calculator that will meet them.
    </p>
    <br />
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione, autem est odio cumque commodi voluptatem quam labore eaque officiis molestiae, dignissimos voluptate magni quasi exercitationem velit harum suscipit natus vel. Eum, soluta doloribus! Molestias dolorem corporis, soluta mollitia tempore vel totam quibusdam dolores aut, eaque itaque veritatis enim alias eum excepturi voluptates ea cupiditate doloribus. Aspernatur quasi voluptatem est sunt. Ducimus cupiditate dicta illo, fugit est unde eos maiores molestias! Quod iste nulla consectetur natus dolorum modi vero suscipit libero neque, expedita mollitia ea explicabo, placeat et, inventore iusto aspernatur?
    </p>
  </div>
</div>
`;
